sdks/java/io/hadoop-format/src/main/java/org/apache/beam/sdk/io/hadoop/format/HadoopFormatIO.java

```diff
@@ -197,99 +425,764 @@ public void validate(PipelineOptions pipelineOptions) {}
     @Override
     public void populateDisplayData(DisplayData.Builder builder) {
       super.populateDisplayData(builder);
-      Configuration hadoopConfig = getConfiguration().get();
+      Configuration hadoopConfig = getConfiguration();
       if (hadoopConfig != null) {
         builder.addIfNotNull(
-            DisplayData.item(OUTPUTFORMAT_CLASS, hadoopConfig.get(OUTPUTFORMAT_CLASS))
+            DisplayData.item(OUTPUT_FORMAT_CLASS_ATTR, hadoopConfig.get(OUTPUT_FORMAT_CLASS_ATTR))
                 .withLabel("OutputFormat Class"));
         builder.addIfNotNull(
-            DisplayData.item(OUTPUTFORMAT_KEY_CLASS, hadoopConfig.get(OUTPUTFORMAT_KEY_CLASS))
+            DisplayData.item(OUTPUT_KEY_CLASS, hadoopConfig.get(OUTPUT_KEY_CLASS))
                 .withLabel("OutputFormat Key Class"));
         builder.addIfNotNull(
-            DisplayData.item(OUTPUTFORMAT_VALUE_CLASS, hadoopConfig.get(OUTPUTFORMAT_VALUE_CLASS))
+            DisplayData.item(OUTPUT_VALUE_CLASS, hadoopConfig.get(OUTPUT_VALUE_CLASS))
                 .withLabel("OutputFormat Value Class"));
+        builder.addIfNotNull(
+            DisplayData.item(
+                    PARTITIONER_CLASS_ATTR,
+                    hadoopConfig.get(
+                        PARTITIONER_CLASS_ATTR,
+                        HadoopUtils.DEFAULT_PARTITIONER_CLASS_ATTR.getName()))
+                .withLabel("Partitioner Class"));
       }
     }
 
     @Override
-    public PDone expand(PCollection<KV<K, V>> input) {
-      input.apply(ParDo.of(new WriteFn<>(this)));
-      return PDone.in(input.getPipeline());
+    public PDone expand(PCollection<KV<KeyT, ValueT>> input) {
+
+      // streamed pipeline must have defined configuration transformation
+      if (!input.isBounded().equals(PCollection.IsBounded.BOUNDED)
```

Comment:
i guess this could be simplified to
```
       if (input.isBounded().equals(PCollection.IsBounded.UNBOUNDED) 
```

the we should add check for the windowing strategy (it should not use global window)